### DISCUSSION TITLE
Error en Git

### DISCUSSION THREAD

**Discussion by BeKuKeSa** [Launch-X-Latam/MisionFrontEnd#178] [Q&A] [answered]
Buenas tardes compañeros, tengo un problema al querer subir mis archivos a GitHub, me sale el siguiente error:

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address

La verdad soy completamente nuevo en esto me podrían orientar para saber que hago mal, gracias :)

**Comment by JesusTriana18** [comment]
Hola, puedes probar haciendo estos pasos: https://www.enmimaquinafunciona.com/pregunta/140686/como-configuro-mi-identidad-git-local . A mi me funciono cuando empece a usar git :)

**Comment by EsthebanDiego** [comment]
La primera vez que usas Git para subir los repositorios tienes que  setear tu usuario y tu correo
en la terminal tienes que poner
git config --global user.name "entre parentesis va el usuario que tienes en github" y das enter
luego
git config --global user.email "entre parentesis el email con el que te diste de alta en github" y das enter
ya con eso deberias de poder subir tus archivos a un repositorio

**Comment by doguedogue** [comment]
Están bien tus comando pero no uses comillas mas que para el nombre si tiene espacios. Estos comando los ejecutas en una terminal CMD, PowerShell o linux es igual.

Te introduce tu email global para git (hay forma de ponerlo local, solo cuando hay mas usuarios en tu equipo)
`git config --global user.email juan.perez@example.com`

Te introduce tu nombre global para git
`git config --global user.name "Juan Pérez"`

Te lista la configuración actual, si no aparece completo en el 
`git config -l`

Te lista los logs de los commits hechos (usa las flechas arriba y abajo y la letra "q" para salir del modo validación de logs)
`git logs`

Listo, con eso ya tienes para ver subas tu repositorio con tu nombre correctamente en github.

**Reply by BeKuKeSa** [reply]
Muchas gracias a todos, así lo intentaré

**Comment by Amhedriel** [comment, marked as answer]
Buenas BeKuKeSa, espero que hayas podido solucionar tu problema, tengo unos [repositorios](https://github.com/Amhedriel/Mark_Down_Tutoriales) que te pueden ayudar un poco con Git, el Bash de Git y estoy probando con GitHub, lo estoy editando en MarkDown para que pueda ser leído más fácil en GitHub.

Saludos.

**Reply by BeKuKeSa** [reply]
Muchas gracias por el apoyo, intente de la manera que compartiste pero ahora tengo este error al querer subir mis archivos 

On branch main
nothing to commit, working tree clean

Reinstale Git y nada, no se que hago mal

**Reply by Amhedriel** [reply]
Buenas de nuevo, perdon por el retraso en contestar, trabajo 12 horas/día y me falta tiempo =P.
`On branch main` quiere decir que la `rama` (branch que quire decir Ramificación) en la que estás actualmente es la `principal` (main/master)

`nothing to commit` (working tree clean nada que confirmar, árbol de trabajo limpio), quiere decir que no tienes ningún cambio/ modificación realizada en tu rama y si no tienes ningún cambio para subir a tu rama entonces no tiene sentido intentar subir "algo", intenta modificar algún archivo agregando o borrando algo de texto para poder commitear (esta palabra no existe pero se entiende XD) y si aún así no te deja subir nada consulta de nuevo la guía, aunque los consejos que te dan los colegas son correctos también.

Reinstalar Git no es necesario, porque git no es un programa y nada mas, como sería una App, git es especial: Git es un software de control de versiones diseñado por Linus Torvalds, pensando en la eficiencia, la confiabilidad y compatibilidad del mantenimiento de versiones de aplicaciones cuando estas tienen un gran número de archivos de código fuente. En resumen te diría que Git no falla, el problema somos nosotros, por commitear mal, hacer merge a algún archivo que no queríamos subir para que no falle nada, ingresamos un bug al `main` y nos cargamos el software que estábamos desarrollando en nuestro repositorio ToT.

Cualquier problema avisas y lo vemos, abrazos

**Reply by BeKuKeSa** [reply]
No hay problema, muchas gracias aun hay errores pero sigo investigando, tu repositorio me ayudo mucho

**Comment by EsthebanDiego** [comment]
normalmente cuando creas tu primer repositorio estas en la branch `master`, pero en las intrucciones de git normalmente viene que lo subas al branch `main`
observa bien como sale en tu terminal la parte entre parentesis si dice (master) o (main)
ya dependiendo haces el push con lo que diga entre los parentesis

`git push origin master`
`git push origin main`

**Reply by BeKuKeSa** [reply]
Gracias por la info, en mi caso es main pero al ejecutar el comando sale lo siguiente:

$ git push origin main
fatal: protocol '?[200~https' is not supported

**Comment by EsthebanDiego** [comment]
No estoy seguro del por qué ese error, parece que es un problema cuando añades el repositorio externo, igual a mí me sirvió el siguiente `video tutorial de git` [GIT y GITHUB para principiantes - Tutorial 2022](https://www.youtube.com/watch?v=Fw4wxIFCpP8&t=3737s&ab_channel=TheFullstackDevs)

**Reply by BeKuKeSa** [reply]
Muchas gracias, lo revisare
